crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -191,9 +231,21 @@ impl<'w> EntityMut<'w> {
 
     #[inline]
     pub fn get<T: Component>(&self) -> Option<&'_ T> {
-        // SAFETY: lifetimes enforce correct usage of returned borrow
+        // SAFETY:
+        // - entity location is valid
+        // - archetypes and components come from the same world
+        // - world access is immutable, lifetime tied to `&self`
         unsafe {
-            get_component_with_type(self.world, TypeId::of::<T>(), self.entity, self.location)
+            self.world
+                .storages
+                .get_component_with_type(
+                    &self.world.archetypes,
```

Comment:
Almost all of these are using borrows from `World`. Wouldn't it make more sense to move these to `World` than `Storages`?

Reply:
If it cannot be safely used without them, it's best to put it on the interface where it's most cleanly used and has fewer safety invariants.